packages/dcos-integration-test/extra/test_dcos_diagnostics.py

```diff
@@ -262,10 +264,17 @@ def get_nodes_from_response(response):
 
         agent_nodes_response = check_json(
             dcos_api_session.health.get('/units/dcos-mesos-slave.service/nodes', node=master))
-
         agent_nodes = get_nodes_from_response(agent_nodes_response)
 
-        assert len(agent_nodes) == len(dcos_api_session.slaves), '{} != {}'.format(agent_nodes, dcos_api_session.slaves)
+        # Fetch Windows nodes if we have some.
+        if 'WinRM' in pulled_units:
+            windows_agent_nodes_response = check_json(
+                dcos_api_session.health.get('/units/mesos-agent/nodes', node=master))
+            windows_agent_nodes = get_nodes_from_response(windows_agent_nodes_response)
+        else:
+            windows_agent_nodes = list()
+
+        assert set(agent_nodes + windows_agent_nodes) == set(dcos_api_session.slaves)
```

Comment:
can't we add this to test utils?

Reply:
It's not a easy change. We should change the way how we start tests and add new a env variable with windows private nodes. Then we can use it to distinguish IPs in tests.

Reply:
Ideally the tests would be mostly agnostic to the node types. Eg the service units should have the same name. I think it is ok to make it explicit right now and not hide it in a library.